pkg/planner/core/operator/logicalop/base_logical_plan.go

```diff
@@ -35,6 +35,11 @@ import (
 
 var _ base.LogicalPlan = &BaseLogicalPlan{}
 
+const (
+	// ApplyGenFromXFDeCorrelateRuleFlag is the flag marked for this op apply is intermediary.
+	ApplyGenFromXFDeCorrelateRuleFlag uint64 = 1 << 0
```

Comment:
or put the example here

Reply:
I will add a concrete exmple test case at following pr